### ISSUE TITLE
Empty Content-Type Header in APIgee Management API request

### ISSUE THREAD

**Issue by eugenebotov** [scastria/terraform-provider-apigee#9]
Hello, I'm using provider 0.1.32 (run with terraform Docker image [0.13.5](https://hub.docker.com/r/hashicorp/terraform)) and APIgee Private Cloud 4.19.01 and facing following issue:

any request returns a 400 Bad Request Error:

`Error: Status 400: Message: Bad Request: {
"code" : "rest.InvalidContentType",
"message" : "Invalid content type ",
"contexts" : [ ],
"cause" : {
"message" : "Media type separator is missing",
 "contexts" : [ ]
 }`

With trace log level I can see that there is a Content-Type Header in the request with an empty value:

`2021-02-02T08:36:37.080Z [DEBUG] plugin.terraform-provider-apigee_v0.1.32: Host: [MASKED]:8443`
`2021-02-02T08:36:37.080Z [DEBUG] plugin.terraform-provider-apigee_v0.1.32: Authorization: Basic [MASKED]`
`2021-02-02T08:36:37.080Z [DEBUG] plugin.terraform-provider-apigee_v0.1.32: Content-Type: `
`2021-02-02T08:36:37.080Z [DEBUG] plugin.terraform-provider-apigee_v0.1.32: `

and I can reproduce it if I send a request with an empty Content-Type header manually:

`curl --request POST \
  --url 'https://URL/v1/o/org/apis?action=import&name=name&validate=true' \
  --header 'authorization: Basic token' \
  --header 'content-type: '
`
the response is 
`{
  "code": "rest.InvalidContentType",
  "message": "Invalid content type ",
  "contexts": [
  ],
  "cause": {
    "message": "Media type separator is missing",
    "contexts": [
    ]
  }
}`

Is there any way to specify content type explicitly? Couldn't find anything on this in the docs.

**Comment by scastria** [comment]
Strange.  I will see if I can reproduce and get back with you.  I may need you to help me reproduce if I have trouble.

**Comment by scastria** [comment]
I pulled the docker image.  What command did you use to run it?  Are you copying your TF template inside the container or mounting a volume so it can use your TF template on the host machine?

**Comment by eugenebotov** [comment]
I use the image inside GitLab CI pipeline, so the TF template is copied inside the container. The container's entrypoint is overridden by GitLab as shown [here](https://docs.gitlab.com/ee/ci/docker/using_docker_images.html#overriding-the-entrypoint-of-an-image), and after that the commands run are

`terraform init -get-plugins=false`
`terraform plan`
`terraform apply -auto-approve`

**Comment by scastria** [comment]
I can see where the content type could be empty since its value is the return value of a function as shown here:

	requestHeaders := http.Header{
		headers.ContentType: []string{mp.FormDataContentType()},
	}

So let me see if I can figure out why mp.FormDataContentType() is returning empty

**Comment by scastria** [comment]
Oh, I forgot to say that I cannot reproduce.  I mounted a volume to my container so that I could reference my template and proxy bundle zip file and I was able to import the proxy just fine.

**Comment by scastria** [comment]
Can you post your TF template, just the section for this failing proxy resource?  An example would be:

resource "apigee_proxy" "MyProxy" {
  name = "hello-world"
  bundle = "proxies/hello-world/hello-world.zip"
  bundle_hash = filebase64sha256("proxies/hello-world/hello-world.zip")
}

**Comment by scastria** [comment]
While I am researching, can you try the following on your local machine to see what happens:

1. Download the attached hello-world proxy bundle zip file
2. Create a TF template that has this resource defined:
resource "apigee_proxy" "MyProxy" {
    name = "hello-world"
    bundle = "hello-world.zip"
    bundle_hash = filebase64sha256("hello-world.zip")
}
3. From your local machine that contains both the TF template and the hello world zip file run docker while mounting your current directory to a /host directory inside the container (change `pwd` in case you are running on windows with something equivalent)
docker run -it --rm --entrypoint /bin/sh --volume `pwd`:/host hashicorp/terraform:0.13.5
4. Now you are inside your container running a shell
5. cd /host
6. terraform init
7. terraform apply -auto-approve
8. This works for me.  What happens for you?

[hello-world.zip](https://github.com/scastria/terraform-provider-apigee/files/5921845/hello-world.zip)

**Comment by scastria** [comment]
Looking at the source code for GoLang itself (the language a terraform provider must be written in), I do NOT see any way that mp.FormDataContentType() can return an empty string.  I am going to wait for your responses from my messages above as I don't know what else to do.

I am hoping you are just providing a bad file in the bundle property of the TF template.

**Comment by eugenebotov** [comment]
> While I am researching, can you try the following on your local machine to see what happens:
...
> 8. This works for me.  What happens for you?

that works for me too. however, if I add another resource to the same template it cannot be created with the same error:

apigee_target_server.example: Creating...
apigee_proxy.MyProxy: Creating...
apigee_proxy.MyProxy: Creation complete after 5s [id=hello-world]

Error: Status 400: Message: Bad Request: {
  "code" : "rest.InvalidContentType",
  "message" : "Invalid content type ",
  "contexts" : [ ],
  "cause" : {
    "message" : "Media type separator is missing",
    "contexts" : [ ]
  }
}

resources in my template are:
resource "apigee_target_server" "example" {
  environment_name = "dev"
  name = "Authentication"
  host = "auth.company.com"
  port = 80
}

resource "apigee_proxy" "MyProxy" {
name = "hello-world"
bundle = "hello-world.zip"
bundle_hash = filebase64sha256("hello-world.zip")
}

The very first template that I tried and that also threw 400 Bad request was this:

resource "apigee_environment_kvm" "test" {
  environment_name = "dev"
  name = "tftest"
  entry = {
    first = "firstValue"
    second = "secondValue"
  }
}

**Comment by scastria** [comment]
I can reproduce the Media type separator is missing issue when trying to create an environment KVM.  That should help me figure out what is wrong.

**Comment by scastria** [comment]
I discovered that the mime package I was using to generate ContentTypes (so that I wouldn't have to hardcode them) is dependent on the underlying OS known mime types.  Therefore, the TF docker container probably doesn't have any known mime types causing null strings.  Therefore, I switched all ContentTypes to hardcoded values (except for when uploading a multipart file, like for proxies) and I was able to fix the environment KVM issue.

So try out v0.1.33 and see if it works better.
